syft/source/content_requester.go

```diff
@@ -0,0 +1,48 @@
+package source
+
+import "sync"
+
+type ContentRequester struct {
+	request map[Location][]*FileData
+	lock    sync.Mutex
+}
+
+func NewContentRequester(data ...*FileData) *ContentRequester {
```

Comment:
It looks like arguments aren't being supplied to this function. Do you see an immediate case where they will be? Would it make sense to defer the idea of taking parameters until they're needed?

Reply:
Convenience for allowing something like `NewContentRequester(a_slice...).Execute(resolver)`, though this form is not used currently.

Reply:
👍 (We discussed keeping this for now, and seeing if we begin to use this code in the near future.)